PodcastGenerator/core/episodes.php

```diff
@@ -57,34 +95,44 @@ function getEpisodes($category = null, $_config)
                     continue;
                 }
             }
-            $append_array = [
-                'episode' => [
-                    'titlePG' => $item->titlePG,
-                    'shortdescPG' => $item->shortdescPG,
-                    'longdescPG' => $item->longdescPG,
-                    'imgPG' => $item->imgPG,
-                    'categoriesPG' => [
-                        'category1PG' => $item->categoriesPG->category1PG,
-                        'category2PG' => $item->categoriesPG->category2PG,
-                        'category3PG' => $item->categoriesPG->category3PG
-                    ],
-                    'keywordsPG' => $item->keywordsPG,
-                    'explicitPG' => $item->explicitPG,
-                    'authorPG' => [
-                        'namePG' => $item->authorPG->namePG,
-                        'emailPG' => $item->authorPG->emailPG
-                    ],
-                    'fileInfoPG' => [
-                        'size' => $item->fileInfoPG->size,
-                        'duration' => $item->fileInfoPG->duration,
-                        'bitrate' => $item->fileInfoPG->bitrate,
-                        'frequency' => $item->fileInfoPG->frequency
-                    ],
-                    'filename' => $episode,
-                    'fileid' => pathinfo($_config['absoluteurl'] . $_config['upload_dir'] . $episode, PATHINFO_FILENAME),
-                    'moddate' => date('Y-m-d', filemtime($_config['absoluteurl'] . $_config['upload_dir'] . $episode))
-                ]
-            ];
+            $append_array = arrayEpisode($item, $episode, $_config);
+            array_push($episodes_data, $append_array);
+        }
+    }
+    unset($_config);
+    return $episodes_data;
+}
+
+function searchEpisodes($name = "", $_config)
+{
+    $name = strtolower($name);
+    $episodes_mtimes = setupEpisodes($_config);
+    // Check if name is a category and replace
+    $cats_xml = simplexml_load_file('categories.xml');
+    foreach ($cats_xml as $item) {
+        if ($name === strtolower($item->description[0])) {
+            $name = strval($item->id[0]);
+        }
+    }
+    // Get XML data for the episodes of interest.
+    $episodes_data = array();
+    for ($i = 0; $i < sizeof($episodes_mtimes); $i++) {
+        $episode = $episodes_mtimes[$i][0];
```

Comment:
Is this a good way to access the episode name? I would suggest accessing it through the key directly as this is more descriptive.

Reply:
Sorry, I don't understand what you mean, note the $episode var is used twice.

Reply:
I mean do we need the 0 index or can we just use "key"?
I might understand it wrong as well

Reply:
Yes it is required as $episodes_mtimes is a 2D array with index 0 as $entryand and index 1 as $this_mtime as seen in the function setupEpisodes($_config).